File: projects/11_rs232_rx/matlab_codes/rs232_protocol_rx.m
%% Clean up
clear; close all; clc;

%% Open and setup COM port
com = "COM12";
baud = 9600;
fpga = serialport(com,baud,"Timeout",10,"Parity","none");

%% Write data
value = 127;
write(fpga,value,"uint8");

%% Close COM port
clear fpga;

File: projects/11_rs232_rx/python_codes/rs232_protocol_rx.py
import serial

fpga = serial.Serial(
    port          = 'COM12',
    baudrate      = 9600,
    bytesize      = serial.EIGHTBITS,
    parity        = serial.PARITY_EVEN,
    stopbits      = serial.STOPBITS_ONE, 
    timeout       = 10,
    write_timeout = 10
)

value = b'\xaa'
fpga.write(value)

fpga.close()